index.js

```diff
@@ -178,25 +183,16 @@ function getHeaders (registry, uri, opts) {
   if (opts.npmCommand)
     headers['npm-command'] = opts.npmCommand
 
-  const auth = getAuth(registry, opts)
+  const auth = getAuth(uri, opts)
   // If a tarball is hosted on a different place than the manifest, only send
   // credentials on `alwaysAuth`
-  const shouldAuth = (
-    auth.alwaysAuth ||
-    new url.URL(uri).host === new url.URL(registry).host
-  )
-  if (shouldAuth && auth.token)
+  if (auth.token)
```

Comment:
so we're deprecating the `always-auth` flag in favor of sending auth whenever it's available for the registry specified at the given host? just want to be certain that's intentional.

Reply:
Yeah, I'm not sure how we get around that, tbh.  We could of course land this PR, and then bring always-auth back and only send if the package is scoped, but I suspect that getting the wrong value for `shouldAuth` is the source of a lot of our 4xx errors with private registries.

Reply:
From the commit message:

    BREAKING CHANGES:

    * The `alwaysAuth` config is no longer relevant.  If we have
      authentication information for a given URI, we send it when making the
      request to that URI.  This was previously being done based on the
      presence of a `scope` parameter or a scoped `spec` option.  However,
      that is not reliable, as users can depend directly on tarball URLs,
      and other parts of the system may have reason to make requests to the
      registry in question without a specific dependency spec in mind (for
      example, to fetch a packument for the purpose of calculating
      metavulnerabilities during an `npm audit` operation.)

    * A top level `_auth`, `_authToken`, `username`, `_password`, or
      `password` option is no longer respected if not scoped to a given
      registry URL.  This functionality was removed from npm/cli as of
      version 7, due to the risk of sending credentials to an incorrect host
      by mistake.  The npm cli will automatically scope these top-level
      configs to the default registry option and save them back to the
      `.npmrc` file the first time they are encountered, so users generally
      see no disruption, and this change only affects non-npm users of
      npm-registry-fetch.